posthog/api/web_experiment.py

```diff
@@ -0,0 +1,184 @@
+            if transforms is None:
+                raise ValidationError(f"Experiment variant '{name}' does not have any element transforms")
+            for idx, transform in enumerate(transforms):
+                if transform.get("selector") is None:
+                    raise ValidationError(
+                        f"Experiment transform [${idx}] variant '{name}' does not have a valid selector"
+                    )
+
+        return attrs
+
+    def create(self, validated_data: dict[str, Any]) -> Any:
+        create_params = {
+            "name": validated_data.get("name", ""),
+            "description": "",
+            "type": "web",
+            "variants": validated_data.get("variants", None),
+            "filters": {
+                "events": [{"type": "events", "id": "$pageview", "order": 0, "name": "$pageview"}],
+                "layout": "horizontal",
+                "date_to": "2024-09-05T23:59",
+                "insight": "FUNNELS",
+                "interval": "day",
+                "date_from": "2024-08-22T10:44",
+                "entity_type": "events",
+                "funnel_viz_type": "steps",
+                "filter_test_accounts": True,
+            },
+        }
+
+        filters = {
+            "groups": [{"properties": [], "rollout_percentage": 100}],
+            "multivariate": self.get_variant_names(validated_data),
+        }
+
+        feature_flag_serializer = FeatureFlagSerializer(
+            data={
+                "key": f'{validated_data.get("name")}-feature',
+                "name": f'Feature Flag for Experiment {validated_data["name"]}',
+                "filters": filters,
+                "active": False,
+            },
+            context=self.context,
+        )
+
+        feature_flag_serializer.is_valid(raise_exception=True)
+        feature_flag = feature_flag_serializer.save()
+
+        experiment = Experiment.objects.create(
+            team_id=self.context["team_id"], feature_flag=feature_flag, **create_params
+        )
+        return experiment
+
+    def update(self, instance: WebExperiment, validated_data: dict[str, Any]) -> Any:
+        variants = validated_data.get("variants", None)
+        if variants is not None and isinstance(variants, dict):
+            feature_flag = instance.feature_flag
+            filters = {
+                "groups": feature_flag.filters.get("groups", None),
+                "multivariate": self.get_variant_names(validated_data),
+            }
+
+            existing_flag_serializer = FeatureFlagSerializer(
+                feature_flag,
+                data={"filters": filters},
+                partial=True,
+                context=self.context,
+            )
+            existing_flag_serializer.is_valid(raise_exception=True)
+            existing_flag_serializer.save()
+
+        instance = super().update(instance, validated_data)
+        return instance
+
+    def get_variant_names(self, validated_data: dict[str, Any]):
```

Comment:
This method retrieves variants and converts them to the format used by FeatureFlag. I suggest making this clearer by either renaming it to `get_variants_for_feature_flag` or adding a comment.